tools/buildsys/src/builder.rs

```diff
@@ -367,8 +459,10 @@ impl DockerBuild {
             &self.state_dir,
         )?;
 
-        // Clean up any previous outputs we have tracked.
-        clean_build_files(&marker_dir, &self.artifacts_dir)?;
+        // Clean up any previous outputs we have tracked, unless we are repacking.
+        if self.target != "repack" {
+            clean_build_files(&marker_dir, &self.artifacts_dir)?;
+        }
```

Comment:
It might be more aligned with the new structure to have a `CommonBuildArgs` with a `clean_output_before_build` flag, which could then be set to false. Otherwise, better to use match on `TargetBuildArgs` to control this, vs string matching on targets which could break if the `Dockerfile` is refactored.

Can you verify that the right marker files are created under `build/state` for the files generated by your build? We want the property that a resign operation followed by a clean variant build with remove the files created during resign, which is what the markers accomplish.